movie_search/views.py

```diff
@@ -240,20 +241,82 @@ def search(request):
 
         return render(request, "media_search.html", context)
 
+def get_movie_genres(movie):
+    
+    movie_genres = []
 
-def movie_detail(request, obj_id):
-
-    movie_detail = media_api.get_media_data(f"/movie/{obj_id}")
-    # pprint("MOVIE DETAIL: ", movie_detail)
+    genres = movie.get("genres")
 
-    movie_videos = media_api.get_media_data(f"/movie/{obj_id}/videos")
+    if genres is not None:
+        for row in genres:
+            genre, inserted = Genre.objects.get_or_create(
```

Comment:
this create objects in a loop, you could do one bulk create as well

Reply:
So I tried adding this:
```
def get_movie_genres(movie):
    
    movie_genres = []

    genres = movie.get("genres")

    if genres is not None:
        for row in genres:
            genre = Genre(name=row.get("name", ""),genre_id=row.get("id", ""),)
            movie_genres.append(genre)
        Genre.objects.bulk_create(movie_genres)
    
    return movie_genres
```

But I encountered this error:

![image](https://github.com/vnoelifant/movie-search-app/assets/43191157/5d36f9e3-a1a5-43bc-a7c0-fc60f1a84a50)

I also have fixtures located in https://github.com/vnoelifant/movie-search-app/tree/47-tv-discover/movie_search/fixtures for the genres and the movie providers. I am confused if I need to load this if doing a Django create/get_or_create.

Reply:
Seems issue with database setting, the code looks good. Let's reproduce and debug this today ...